src/autoqasm/types/types.py

```diff
@@ -105,11 +107,13 @@ def __init__(self, *args, annotations: str | Iterable[str] | None = None, **kwar
                 "Arrays may only be declared at the root scope of an AutoQASM main function."
             )
 
-        if not args:
-            raise errors.InvalidArrayDeclaration("init_expression must at least be provided.")
```

Comment:
I think it'd be good to keep the `InvalidArrayDeclaration` if `init_expression` is not provided (or is not an iterable).

Reply:
Sure. Since `init_expression` must be an iterable type, I have added that check. But Python's `ValueError` is raised if `init_expression` is not provided. Do you think this is fine?

Reply:
Yes, that makes sense to me. Thank you for making the change!